am-lib/src/main/java/uk/gov/hmcts/reform/amlib/AccessManagementService.java

```diff
@@ -45,21 +52,36 @@ public void createResourceAccess(ExplicitAccessRecord explicitAccessRecord) {
     }
 
     /**
-     * Returns `resourceJson` when record with userId and resourceId exist and has READ permissions, otherwise null.
+     * Returns FilterResourceResponse when record with userId and resourceId exist and has READ permissions,
+     * otherwise null.
      *
      * @param userId       (accessorId)
      * @param resourceId   resource id
      * @param resourceJson json
      * @return resourceJson or null
+     * @throws UnsupportedPermissionsException when permissions are above 31 or below 0.
      */
-    public JsonNode filterResource(String userId, String resourceId, JsonNode resourceJson) {
+
+    public FilterResourceResponse filterResource(String userId, String resourceId, JsonNode resourceJson)
+        throws UnsupportedPermissionsException {
```

Comment:
Please remove declaration of runtime exceptions. They are not required by the language and make code less readable. Remember to update Javadoc as well.

See: https://docs.oracle.com/javase/tutorial/essential/exceptions/runtime.html

Reply:
👍